server/routes/event.js

```diff
@@ -5,38 +5,43 @@ import eventController from '../controllers/event'
 import eventValidation from '../validation/event'
 
 import { canReadAsVisitor, canCreate, canUpdate, canDelete } from '../middlewares/roles'
-import { checkTicketStatus } from '../middlewares/status'
+import { grantWhenFestivalFree } from '../middlewares/authorizeFreeFestival'
+import { authorizeJWT } from '../middlewares/authorizeJWT'
+import { checkUser } from '../middlewares/checkUser'
 
 const router = express.Router() // eslint-disable-line new-cap
 
 router.route('/')
   .get(
-    checkTicketStatus,
-    canReadAsVisitor,
+    checkUser,
```

Comment:
Do we really need `checkUser` here? Is it required by the controller?

`grantWhenFestivalFree` also does not need to have a user object, it just needs to know if one exists.

Reply:
I believe it's required in the `base.js` controller to check and set the owner status, here: https://github.com/adzialocha/hoffnung3000/blob/9dc2aac80a63d55e19eb5d5032fd635560789e73/server/controllers/base.js#L45-L50

This means it's needed for the eventController and placeController.... There might be a better way to fix it though, setting req.user in the checkUser middleware is what I came up with.

Reply:
This one serves only as an info for the frontend, you could set it to false when req.user is not set aka `response.isOwnerMe = (req.user && req.user.id === data.animal.userId)`

Reply:
Sorry, maybe I'm missing something obvious ;-p 

But in the case where the `festivalTicketPrice === 0`, but a registered and logged in participant visits the Calendar, then if `user` and therefore `isOwner` isn't set properly then they won't be shown the necessary edit buttons on the front end.... What am I misunderstanding? haha....

Reply:
Good point! Thank you for pointing that out. Then it all actually looks good, I would only refactor the `checkUser` Middleware a little (see comments there)